### PR TITLE
[🐴] Decrement app badge when opening unread chat

### PR DESCRIPTION
**Pull request by haileyok**
- Decrement the badge when opening a chat from the messages list
- Decrement the badge when opening a chat from a push notification
- Increment the badge every time you receive a notification for a chat
	- EXCEPT for if it is the chat you are currently looking at

### CODE REVIEW

File: src/screens/Messages/List/ChatListItem.tsx [hunk 1]
```diff
@@ -103,11 +104,16 @@ function ChatListItemReady({
     setShowActions(true)
   }, [])
 
-  const onPress = useCallback(() => {
+  const onPress = useCallback(async () => {
+    if (convo.unreadCount > 0) {
+      // We only increment the badge once per conversation, so even if there are multiple unreads
+      // we only want to decrement by one.
+      decrementBadgeCount(1)
```
Comment (estrattonbailey) [inline src/screens/Messages/List/ChatListItem.tsx:111]: We're doing `markAsRead` within the `ConvoProvider`, could be a good spot for this too

Comment (estrattonbailey) [reply src/screens/Messages/List/ChatListItem.tsx:111]: Yeah pry good to do there in case you foreground the app directly to the convo page

Comment (haileyok) [reply src/screens/Messages/List/ChatListItem.tsx:111]: Ahhh okay perfect let me adjust to that, wasn't sure if we did it or the server did it for us.

File: src/lib/hooks/useNotificationHandler.ts [hunk 2]
```diff
@@ -169,10 +169,11 @@ export function useNotificationsHandler() {
           payload.reason === 'chat-message' &&
           payload.recipientDid === currentAccount?.did
         ) {
+          const isCurrentConvo = payload.convoId === currentConvoId
           return {
-            shouldShowAlert: payload.convoId !== currentConvoId,
+            shouldShowAlert: isCurrentConvo,
             shouldPlaySound: false,
-            shouldSetBadge: false,
+            shouldSetBadge: isCurrentConvo,
```
Comment (estrattonbailey) [inline src/lib/hooks/useNotificationHandler.ts:176]: Should these be `!` inverted?

**Review by estrattonbailey — APPROVED** [review]

File: src/lib/hooks/useNotificationHandler.ts (hunk 2 above)
Comment (haileyok) [reply src/lib/hooks/useNotificationHandler.ts:176]: no that doesn't make sense!!! (thanks good catch)

**Pull request merged**